### PR TITLE
wifi install dnsmasq restart issue

### PR DESCRIPTION
**Pull request by jvonau**
#2220 restart fails as br0's creation is being suppressed by not restarting NetworkManager.
Restore preexisting default_vars variable's functionality.

### CODE REVIEW

File: roles/network/defaults/main.yml [hunk 1]
```diff
@@ -91,7 +91,6 @@ systemd_networkd_active: False
 # The values here are default local variables
 gui_wan_iface: "unset"
 gui_static_wan_ip: "unset"
-gui_desired_network_role: Gateway
```
Comment (georgejhunt) [inline roles/network/defaults/main.yml:94]: I think role defaults have the lowest precedence, below ./vars/default_vars, which is below /etc/iiab/local_vars.yml

**Comment by georgejhunt** [comment]
Maybe our documentation should suggest that local_vars.yml.gui_desired_network_role be set to "Appliance" -- so that the install continues to be able to download content even after a reboot.

**Comment by holta** [comment]
Non-technical implementers can now test out this PR, by pasting in this 1-line IIAB installer:

```
curl d.iiab.io/install-test.txt | sudo bash
```

i.e. to install IIAB over a WiFi connection, in situations like #2210 where there is no downstream router/downstream LAN/br0 etc...to make sure reboot etc work the way folks expect!

(Presumably we should test with NetworkManager (i.e. desktop versions of Ubuntu etc) as well as testing OS's that do not use NetworkManager, just to be sure?)

**Comment by m-anish** [comment]
This fixes a corner case if I understand, until the underlying problem is fixed?
Should've been merged?

**Comment by holta** [comment]
This can be merged if that's the right thing to do.

(With at least one fresh install of one kind or another?)

**Comment by holta** [comment]
@jvonau on today's IIAB call (http://minutes.iiab.io) says let's test this PR (with a WiFi install of IIAB) onto any of our 3 main OS's, which I and others will do in the coming weeks.

e.g. I and @avni will try to find an old laptop to test w/ Ubuntu 18.04 Desktop version, to test a scenario just like Victor's.  @jvonau clarifies: this affects all OS's, not just NetworkManager OS's like Ubuntu 18.04, so could likely be tested with just about ANY WiFi install of IIAB.

@georgejhunt & @tim-moody asked some design questions about when br0 can/should exist, re: right after power-outages and reboot during IIAB install & content provisioning/downloading.  (Think of it as Gateway w/o LAN?  Or think of it as appliance?)

**Comment by holta** [comment]
@jvonau @georgejhunt I attempted a WiFi install onto 183-rpi4-2g-lite-0217-BIG-2226 = 10.8.0.26 (with the Ethernet cable unplugged to avoid duplicate route Internet snafus) and this PR looked good up to the point soon after the network stage (roles/network) ran.

Then it failed at `[cmdsrv : Install packages for cmdsrv on python 3]` with this error:

> Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/p/python-systemd/python3-systemd_234-2+b1_armhf.deb  Temporary failure resolving 'raspbian.raspberrypi.org'

Can you tell why `dnsmasq` was not running at that point?  Even despite the fact that the network stage (roles/network) seemed to complete successfully, right after iiab-install's 9 stages?

> root@raspberrypi:~# systemctl status dnsmasq
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
   Loaded: loaded (/lib/systemd/system/dnsmasq.service; disabled; vendor preset: enabled)
   Active: inactive (dead)
>
> Feb 17 02:55:07 box.lan dnsmasq[31989]: Too few arguments.
Feb 17 02:55:07 box.lan systemd[1]: dnsmasq.service: Succeeded.
Feb 17 02:55:07 box.lan systemd[1]: Stopped dnsmasq - A lightweight DHCP and caching DNS server.
Feb 17 02:55:08 box.lan systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
Feb 17 02:55:08 box.lan dnsmasq[32144]: dnsmasq: syntax check OK.
Feb 17 02:55:08 box.lan dnsmasq[32175]: Too few arguments.
Feb 17 02:55:08 box.lan systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.
Feb 17 02:56:46 box.lan systemd[1]: Stopping dnsmasq - A lightweight DHCP and caching DNS server...
Feb 17 02:56:47 box.lan systemd[1]: dnsmasq.service: Succeeded.
Feb 17 02:56:47 box.lan systemd[1]: Stopped dnsmasq - A lightweight DHCP and caching DNS server.

Manually running `systemctl start dnsmasq` worked, but somehow that was not enough to restore very reliable Internet &mdash; which was working fine up until the network stage ran &mdash; whereupon its Internet became unreliable even with dnsmasq manually turned back on (as of now).

If you can log in to investigate that would help.  Its OpenVPN connection restored itself a few minutes later on its own (10.8.0.26) and is working for now.

But just in case you cannot connect to 10.8.0.26, I've published the system's `iiab-diagnostics` here so you at least have a summary:

http://paste.debian.net/1130909/

**Comment by jvonau** [comment]
> @jvonau @georgejhunt I attempted a WiFi install onto 183-rpi4-2g-lite-0217-BIG-2226 = 10.8.0.26 (with the Ethernet cable unplugged to avoid duplicate route Internet snafus) and this PR looked good up to the point soon after the network stage (roles/network) ran.
> 
> Then it failed at `[cmdsrv : Install packages for cmdsrv on python 3]` with this error:
> 
> > Failed to fetch http://raspbian.raspberrypi.org/raspbian/pool/main/p/python-systemd/python3-systemd_234-2+b1_armhf.deb  Temporary failure resolving 'raspbian.raspberrypi.org'
> 
> Can you tell why `dnsmasq` was not running at that point? Even despite the fact that the network stage (roles/network) seemed to complete successfully, right after iiab-install's 9 stages?
> 

easy one, service was shutdown in down-debian.yml and usually restarted later.


> > root@raspberrypi:~# systemctl status dnsmasq
> > ● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
> > Loaded: loaded (/lib/systemd/system/dnsmasq.service; disabled; vendor preset: enabled)
> > Active: inactive (dead)
> > Feb 17 02:55:07 box.lan dnsmasq[31989]: Too few arguments.
> > Feb 17 02:55:07 box.lan systemd[1]: dnsmasq.service: Succeeded.
> > Feb 17 02:55:07 box.lan systemd[1]: Stopped dnsmasq - A lightweight DHCP and caching DNS server.
> > Feb 17 02:55:08 box.lan systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
> > Feb 17 02:55:08 box.lan dnsmasq[32144]: dnsmasq: syntax check OK.
> > Feb 17 02:55:08 box.lan dnsmasq[32175]: Too few arguments.
> > Feb 17 02:55:08 box.lan systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.
> > Feb 17 02:56:46 box.lan systemd[1]: Stopping dnsmasq - A lightweight DHCP and caching DNS server...
> > Feb 17 02:56:47 box.lan systemd[1]: dnsmasq.service: Succeeded.
> > Feb 17 02:56:47 box.lan systemd[1]: Stopped dnsmasq - A lightweight DHCP and caching DNS server.
> 
> Manually running `systemctl start dnsmasq` worked, but somehow that was not enough to restore very reliable Internet — which was working fine up until the network stage ran — whereupon its Internet became unreliable even with dnsmasq manually turned back on (as of now).

Don't know about unreliable after turning back on part but the issue is once dnsmasq was started by captiveportal, resolveconf rewrote /etc/resolv.conf pointing to 127.0.0.1 replacing the ISP nameservers at that point instead of later in network. Now that dnsmasq is running the latest commit won't shut dnsmasq down blindly when not restarting dnsmasq later. Victor hit the code in down-debian.yml as that is called by NM-debian.yml.

I applied the latest commit and ran ./iiab-network should be good to continue with sudo iiab but a clean install would be the true test.

**Comment by holta** [comment]
> I applied the latest commit and ran ./iiab-network should be good to continue with sudo iiab

I ran "sudo iiab" and Admin Console installed cleanly: right now the very final stage of the overall IIAB install (unpacking KA Lite's mandatory en lang pack) is wrapping up, so this one's over the hump and looking great.

Aside: 'sudo iiab' almost immediately rebooted, after apt installed nodejs (12.16.0-1nodesource1) over (12.15.0-1nodesource1), but that's by design (if a bit odd) and luckily the WiFi IP address was the same on reboot.  So I could find it super easy & get back in using ssh &mdash; without even having to use 10.8.0.26 or `ping raspberrypi.local` or nosing around in the router for recent DHCP leases etc!

**Comment by holta** [comment]
> a clean install would be the true test

In that spirit a new/fresh install of this PR (over WiFi) was just begun @ 181-rpi4-1g-lite-0217-MIN-2226 = 10.8.0.54 (feel free to log in & monitor as nec.)

After that I assume it's time to merge this (PR #2226) ?  Unless you feel something else is needed.

**Comment by holta** [comment]
> a new/fresh install of this PR (over WiFi) was just begun @ 181-rpi4-1g-lite-0217-MIN-2226 = 10.8.0.54 (feel free to log in & monitor as nec.)

Success!

(And then `iiab-hotspot-on` presumably needs to be run after reboot, per the [usual instructions](https://github.com/iiab/iiab-factory/blob/master/iiab#L305-L309).)
